common/src/main/java/com/wynntils/features/chat/RevealNicknamesFeature.java

```diff
@@ -123,6 +125,40 @@ public void onPlayerChat(ChatMessageReceivedEvent event) {
         event.setMessage(styledText);
     }
 
+    public static Pair<String, String> getNameAndNick(ChatMessageReceivedEvent event) {
+        AtomicReference<String> username = new AtomicReference<>();
+        AtomicReference<String> nickname = new AtomicReference<>();
+        event.getStyledText().iterate((currentPart, changes) -> {
+            HoverEvent hoverEvent = currentPart.getPartStyle().getStyle().getHoverEvent();
+
+            // If the hover event doesn't exist or it is not SHOW_TEXT event, it's not a nickname text part
+            if (hoverEvent == null || hoverEvent.getAction() != HoverEvent.Action.SHOW_TEXT) {
+                return IterationDecision.CONTINUE;
+            }
+
+            StyledText[] partTexts = StyledText.fromComponent(hoverEvent.getValue(HoverEvent.Action.SHOW_TEXT))
+                    .split("\n");
+
+            for (StyledText partText : partTexts) {
+                Matcher nicknameMatcher = partText.getMatcher(NICKNAME_PATTERN);
+
+                if (nicknameMatcher.matches()) {
+                    nickname.set(nicknameMatcher.group("nick"));
+                    username.set(nicknameMatcher.group("username"));
+                }
+            }
+
+            // If the nickname or username is null, it's not a nickname text part
+            if (nickname.get() == null && username.get() == null) {
+                return IterationDecision.CONTINUE;
+            }
+
+            return IterationDecision.BREAK;
+        });
+
+        return new Pair<>(username.get(), nickname.get());
+    }
+
```

Comment:
I'll spare sending the same message as @ShadowCat117 did in https://github.com/Wynntils/Wynntils/pull/2827/files#r1748046610. TLDR, move this method, but also don't forget to use it in this feature itself. Currently you are just "duplicating" the method, and using it in another feature which breaks feature abstraction.

The code itself also needs to be refined a bit. `StylexText#iterate` is meant to be used for changing parts of a `StyledText`, this case, you are ignoring it. You can use `StyledText` with for-each styled loops as it has `iterator` implemented.

Reply:
I still think it's possible to create some method that can find, parse and return "nickname" StyledTextParts. Although, you'd have to move it to a model now, as `StyledTextUtils` is reserved for special, but not feature specific utility methods. (It's hard to imagine the boundaries, but since you need a nickname regex, we should keep the code in a model).